### PR TITLE
test: E2E Test for CalDav Calendar

### LINKED ISSUE #9372: [CAL-1886] Make E2E test: Can add Office CalDav Calendar
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1886](https://linear.app/calcom/issue/CAL-1886/make-e2e-test-can-add-office-caldav-calendar)</sub>

### PR DESCRIPTION
**Pull request by Ruchir28**
## What does this PR do?
Adds an E2E Test for CalDav Calendar

Fixes #9372
/claim #9372

## Type of change

- [ ] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Are there environment variables that should be set?
  - `E2E_TEST_CALDAV_CALENDER_URL`
  - `E2E_TEST_CALDAV_CALENDAR_USERNAME`
  - `E2E_TEST_CALDAV_CALENDAR_PASSWORD`
  
- What is expected (happy path) to have (input and output)?
   - Run `yarn e2e caldav-calendar.e2e.ts`
   - It should successfully add the CalDav Calendar and test should pass

### CODE REVIEW

File: apps/web/playwright/caldav-calendar.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const CALDAV_CALENDER_URL = process.env.E2E_TEST_CALDAV_CALENDER_URL!;
+const CALDAV_CALENDAR_USERNAME = process.env.E2E_TEST_CALDAV_CALENDAR_USERNAME!;
+const CALDAV_CALENDAR_PASSWORD = process.env.E2E_TEST_CALDAV_CALENDAR_PASSWORD!;
+
+const SHOULD_SKIP_TESTS = !CALDAV_CALENDER_URL || !CALDAV_CALENDAR_USERNAME || !CALDAV_CALENDAR_PASSWORD;
+
+test.describe("CalDav Calendar", () => {
+  // eslint-disable-next-line playwright/no-skipped-test
+  test.skip(SHOULD_SKIP_TESTS, "Skipping due to missing test credentials");
```
Comment (hariombalhara) [inline apps/web/playwright/caldav-calendar.e2e.ts:19]:
```suggestion
  test.skip(SHOULD_SKIP_TESTS, "Skipping 'CalDav Calendar'  tests due to missing CalDav credentials");
```

File: apps/web/playwright/caldav-calendar.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { test } from "./lib/fixtures";
+import { installCaldavCalendar } from "./lib/testUtils";
+
+test.describe.configure({ mode: "parallel" });
+
+test.afterEach(({ users }) => users.deleteAll());
+
+const CALDAV_CALENDER_URL = process.env.E2E_TEST_CALDAV_CALENDER_URL!;
+const CALDAV_CALENDAR_USERNAME = process.env.E2E_TEST_CALDAV_CALENDAR_USERNAME!;
+const CALDAV_CALENDAR_PASSWORD = process.env.E2E_TEST_CALDAV_CALENDAR_PASSWORD!;
```
Comment (hariombalhara) [inline apps/web/playwright/caldav-calendar.e2e.ts:13]: We need to document how one can get all these test credentials and after that we should create these credentials and then merge the PR. cc @zomars @keithwillcode

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Thanks @Ruchir28 🙏
Tested locally till clicking on save button, need details on the credentials.

**Comment by Ruchir28** [comment]
Hey @hariombalhara ,
Regarding credentials, I used this [website](https://fruux.com/)
Here to get the caldav credentials 
1. Login here  [fruux.com](https://fruux.com/)
2. Then in the navbar, click on the Sync button 
3. After that there will be an option to Sync with a new device, click on that
4. Scroll to last there will be an option with  'Other Clients' & 'CalDav' mentioned
5. Complete the Registration
6. After that you will see the following info (User Name, Password, Server URL) which can be used to set the env variables for testing

**Comment by Ruchir28** [comment]
Hey @hariombalhara, 
Added the Requested changes, please review

**Pull request closed without merging**